homework-submission/week5-class16/src/stores/TodosStore.js

```diff
@@ -0,0 +1,174 @@
+import { observable, computed, action, runInAction } from "mobx";
+
+import { configure } from "mobx";
+configure({ enforceActions: "observed" });
+
+const API_Base = "https://hyf-react-api.herokuapp.com";
+
+class TodosStore {
+  @observable
+  edit = false;
+  @observable
+  Data = [];
+  @observable
+  state = "loading";
+
+  @action
+  handleCheck = async todoId => {
+    this.state = "loading";
+    const targetedTodo = this.Data.find(todo => todo._id === todoId);
+    // this code block to show instant change to the user.
+    const newData = this.Data.map(todo => {
+      if (todo._id === todoId) {
+        return { ...todo, done: !todo.done };
+      }
+      return todo;
+    });
+    this.Data = newData;
+    this.state = "done";
+
+    try {
+      await this.patchRequest(todoId, { done: !targetedTodo.done });
+    } catch (error) {
+      runInAction(() => {
+        this.state = "error";
+      });
+    }
+
+    this.getTodos();
+  };
+  @action
+  handleDelete = async todoId => {
+    this.state = "loading";
+
+    try {
+      await this.deleteRequest(todoId);
+      runInAction(() => {
+        this.state = "done";
+      });
+    } catch (error) {
+      runInAction(() => {
+        this.state = "error";
+      });
+    }
+
+    this.getTodos();
```

Comment:
Almost as the previous comment, But you need to change a bit more in the user interface to improve it's functionality.